atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1640,6 +1646,14 @@ private void commitWrites(TransactionService transactionService) {
         });
     }
 
+    private UUID getRequestId(LockToken token) {
+        if (token instanceof LeasedLockToken) {
```

Comment:
As discussed yesterday - please can we not do this, and make the .serverToken() call package private again (as it should always have been)?

Reply:
(we can chat about alternatives, but that method is very much designed to be opaque, so making it public and castable needs some extra refactoring if we're going to do it - instanceof checks should be a flag to you).